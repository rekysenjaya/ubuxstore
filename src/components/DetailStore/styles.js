
import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    title: {
        fontSize: 20,
        fontWeight: 'bold'
    },
    spaceBetween: {
        justifyContent: 'space-between'
    },
    imageHorizontal: {
        width: 100,
        height: 100,
        elevation: 3,
        backgroundColor: '#fff',
        margin: 5,
        padding: 5,
        borderRadius: 10
    },
    font10: {
        fontSize: 10
    }
})

export default styles