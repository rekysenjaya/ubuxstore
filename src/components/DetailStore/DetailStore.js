import React, { Component } from 'react';
import { Text, ScrollView } from 'react-native'
import { Card, CardItem, Icon, View } from 'native-base';
import styles from './styles';
import PropTypes from 'prop-types';


export default class DetailStore extends Component {
    render() {
        const { detail } = this.props

        return (<View>
            <Card>
                <CardItem header bordered>
                    <Text style={styles.title} >{detail.legalName}</Text>
                </CardItem>
                <CardItem bordered>
                    <Icon active name="pin" />
                    <Text>{detail.fullAddress}</Text>
                </CardItem>
                <CardItem bordered>
                    <Icon active name="at" />
                    <Text>{detail.businessEmail}</Text>
                </CardItem>
                <CardItem bordered>
                    <Icon active name="call" />
                    <Text>{detail.acnNumber}</Text>
                </CardItem>
                <CardItem footer style={styles.spaceBetween} >
                    <Text style={styles.font10} >timezone : {detail.timezone}</Text>
                    <Text style={styles.font10} >status : {detail.status}</Text>
                </CardItem>
            </Card>

            <View>
                <Text style={[styles.title, { marginHorizontal: 10 }]} >Images</Text>
                <ScrollView horizontal showsHorizontalScrollIndicator={false} style={{ padding: 5 }} >
                    {Array.isArray(detail.images) && detail.images.map((v, i) => <View key={i} style={styles.imageHorizontal} ><Text>{v.url}</Text></View>)}
                </ScrollView>
            </View>

            <Card>
                <CardItem header >
                    <Text style={styles.title} >Review</Text>
                </CardItem>
                {Array.isArray(detail.review) && detail.review.map((v, i) => (
                    <CardItem key={i} bordered style={styles.spaceBetween} >
                        <View style={{ flexDirection: 'row' }} >
                            <Icon active name="text" />
                            <Text style={{ marginLeft: 10 }} >{v.comment}</Text>
                        </View>
                        <View style={{ alignItems: 'center' }} >
                            <Icon active name="star" />
                            <Text style={{ fontWeight: '900' }} >{v.star}</Text>
                        </View>
                    </CardItem>
                ))}
            </Card>
        </View>)
    }
}

DetailStore.propTypes = {
    detail: PropTypes.object
}