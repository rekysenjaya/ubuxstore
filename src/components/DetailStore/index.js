import DetailStore from './DetailStore';
export default DetailStore;
