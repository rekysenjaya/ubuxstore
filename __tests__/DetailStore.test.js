/**
 * Sample React Native Snapshot Test
 */

'use strict';

import 'react-native';
import React from 'react';
import DetailStore from '../src/components/DetailStore';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.setTimeout(15000);

it('renders correctly', () => {
    const tree = renderer.create(<DetailStore detail={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
});