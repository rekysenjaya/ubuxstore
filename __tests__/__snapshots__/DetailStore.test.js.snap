// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      bordered={true}
      header={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      />
    </View>
    <View
      bordered={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        active={true}
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#000",
                "fontSize": 28,
                "width": 35,
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text />
    </View>
    <View
      bordered={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        active={true}
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#000",
                "fontSize": 28,
                "width": 35,
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text />
    </View>
    <View
      bordered={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        active={true}
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#000",
                "fontSize": 28,
                "width": 35,
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text />
    </View>
    <View
      footer={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomWidth": null,
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          },
          Object {
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 10,
          }
        }
      >
        timezone : 
      </Text>
      <Text
        style={
          Object {
            "fontSize": 10,
          }
        }
      >
        status : 
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          },
          Object {
            "marginHorizontal": 10,
          },
        ]
      }
    >
      Images
    </Text>
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
      style={
        Object {
          "padding": 5,
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      header={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": null,
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Review
      </Text>
    </View>
  </View>
</View>
`;
